src/tictactoe/assets/js/reducers/tictactoeFSM.js

```diff
@@ -0,0 +1,30 @@
+
+import { GameRoundState, GameEvents } from '../actions'
+
+const initialState = {
+    gameState: GameRoundState.NOT_STARTED
+}
+
+const allTransitions = [
+    { event: GameEvents.START_ROUND, from: GameRoundState.NOT_STARTED, to: GameRoundState.WAITING_UPDATE }
+    // WAITING_UPDATE + GET_ROUND_STATE => PLAYER_1_MOVES / PLAYER_2_MOVES / PLAYER_WON / DRAW
+    // PLAYER_1_MOVES + MAKE_MOVE => WAITING_UPDATE
+    // PLAYER_2_MOVES + GET_ROUND_STATE => PLAYER_1_MOVES / PLAYER_2_MOVES / PLAYER_WON / DRAW
+    // PLAYER_WON / DRAW + SHOW_RESULT => ROUND_COMPLETED
+]
+
+const tictactoeFSM = function(state = initialState, action) {
+    var event = action.type
+    var currentState = state.gameState;
+
+    var transition = allTransitions.find(function(t) {
```

Comment:
Thanks, my knowledge of JavaScripts comes from the times where there were only `var`:) Will make the things consistent and get rid of them.

I was approaching it from two directions:
1. I thought that `react-redux` is a ready FSM, but while implementing I was thinking that it is doing not exactly what I want.
2. Then I started to apply my simple implementation of FSM, thinking that `react-redux` will be just needed to propagate the state thought the entire app.

In the end I figured out how `react-redux` is doing what I need and left my custom transitions switch just instead of `switch-case`